Add the JaniHUD

nearlyNonexistent opened the pull request:
🆑nearlyNon
rscadd: The JaniHUD, new glasses selectable by robots and spawning in custodial. They alert the user to messes with a helpful ! icon, allowing you to see tiles that are about to be visibly dirty, or messes hidden under stuff.
/🆑

![image](https://user-images.githubusercontent.com/4041565/44552603-6f08a700-a6f0-11e8-8011-4d73161636f7.png)

atlantiscze commented:
Visually it seems that the HUD icons are layering below objects, but i believe they should instead layer above them - pretty sure thats how all other HUDs do it.

nearlyNonexistent commented:
How can I fix that, though? It's using the same hud_overlay feature the other huds use.

PsiOmegaDelta commented on code/game/objects/effects/decals/cleanable.dm:
```diff
@@ -1,5 +1,9 @@
 /obj/effect/decal/cleanable
 	var/list/random_icon_states
+	var/hud_overlay
+/obj/effect/decal/cleanable/New()
```
`/Initialize()`

PsiOmegaDelta commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -67,3 +67,14 @@
+	icon = 'icons/obj/clothing/glasses.dmi'
+	hud_type = HUD_JANITOR
+
+	New()
```
Relative pathing

PsiOmegaDelta replied:
`/Initialize()`

PsiOmegaDelta commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -67,3 +67,14 @@
+	hud_type = HUD_JANITOR
+
+	New()
+		..()
```
`. = ..()`

PsiOmegaDelta commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -67,3 +67,14 @@
+	New()
+		..()
+		hud = new /obj/item/clothing/glasses/hud/janitor(src)
+		return
```
Unnecessary return.

babydoll commented:
Science goggle/HUD things are already purple

nearlyNonexistent commented:
What alt color should I use? Janitor jumpsuits are purple.

babydoll commented:
I guess it's not really a major priority, but you could make it brown or beige or something

SierraKomodo commented:
Yellow could work? - Same color as the ! icons.

nearlyNonexistent commented:
![image](https://user-images.githubusercontent.com/4041565/44960554-4ab68280-aec7-11e8-9bf1-30e6a30671cb.png)

this is ready for review now, cleaned up some borg code while I was at it

nearlyNonexistent committed:
Adds the JaniHUD

🆑nearlyNon
rscadd: The JaniHUD, new glasses selectable by robots and spawning in custodial. They alert the user to messes with a helpful ! icon, allowing you to see tiles that are about to be visibly dirty, or messes hidden under stuff.
/🆑

nearlyNonexistent commented:
someone review this please

babydoll commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -63,7 +62,17 @@
-	New()
-		..()
+/obj/item/borg/sight/hud/Initialize()
+		. = ..()
```
This block is double indented

babydoll commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -63,7 +62,17 @@
+	hud_type = HUD_JANITOR
+
+/obj/item/borg/sight/hud/jani/Initialize()
+		. = ..()
```
Also double indented

babydoll commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -51,10 +51,9 @@
-	New()
-		..()
+/obj/item/borg/sight/hud/med/Initialize()
+		. = ..()
```
Double indents. Did you check to see if the conversion from New to Init.. works?

nearlyNonexistent replied:
It works as far as I can tell. Fixing indents in a bit.

babydoll commented on code/modules/clothing/glasses/hud.dm:
```diff
@@ -60,5 +60,23 @@
+
+/obj/item/clothing/glasses/hud/janitor
+	name = "janiHUD"
+	desc = "Detects FOREIGN CONTAMINANT."
```
`contaminants` probably

babydoll replied:
Also not really as descriptive as the other descs

babydoll commented on code/modules/clothing/glasses/hud.dm:
```diff
@@ -60,5 +60,23 @@
+	name = "prescription janiHUD"
+	icon_state = "janihudpresc"
+	item_state = "glasses"
+	desc = "A janitor HUD integrated with a set of prescription glasses"
```
Needs punctuation

babydoll commented on code/modules/client/preference_setup/loadout/lists/eyegear.dm:
```diff
@@ -111,4 +111,13 @@
 	path = /obj/item/clothing/glasses/sunglasses/blindfold
-	flags = GEAR_HAS_COLOR_SELECTION
\ No newline at end of file
+	flags = GEAR_HAS_COLOR_SELECTION
```
Needs a new line after this

nearlyNonexistent committed:
Merge branch 'dev' into janiHUD
